game/view/pickwidget.lua

```diff
@@ -49,23 +52,31 @@ function PickWidgetView:draw()
 
   g.translate(_width/8, _height/2-2*(_BLOCK_HEIGHT+_MARGIN))
   -- draw stuff
+  local strs = {}
+  local width = 0
   for index, widget in self.target:getBody():eachWidget() do
-    local name, pd
-    name = widget:getName()
+    strs[index] = _FMT:format(widget:getName(),
+                              widget:getWidgetCharges() - widget:getUsages(),
+                              widget:getWidgetCharges(),
+                              _TRIGGERS[widget:getWidgetTrigger()] or ""
+    )
+    width = math.max(width, _font:getWidth(strs[index]))
+  end
+  for index, info_str in ipairs(strs) do
     local selected = self.selection == index
     if selected then
       g.setColor(0xff, 0xff, 0xff, self.alpha*0xff)
     else
       g.setColor(0x16, 0x16, 0x16, self.alpha*0xff)
     end
-    g.rectangle("fill", 0, 0, _text_width, _BLOCK_HEIGHT)
+    g.rectangle("fill", 0, 0, width+8*_PADDING, _BLOCK_HEIGHT)
```

Comment:
This makes the rectangle fir to the text. In particular, if you only have a "Sord", it becomes quite small. Is this intended? (just checking, I'm OK either way)

Reply:
yes, this also makes the rectangle fit for larger names, like the stash buffs